### PR TITLE
add nansum api to math

### PR DESCRIPTION
**Pull request by esythan**
### PR types
New features

### PR changes
APIs

### Describe
add nansum api
```
x = np.array([[float('nan'), 0.3, 0.5, 0.9],  
                [0.1, 0.2, float('-nan'), 0.7]]).astype(np.float32)
x = paddle.to_tensor(x)
out1 = paddle.nansum(x)  # [2.7]
out2 = paddle.nansum(x, axis=0)  # [0.1, 0.5, 0.5, 1.6]
out3 = paddle.nansum(x, axis=-1)  # [1.7, 1.0]
out4 = paddle.nansum(x, axis=1, keepdim=True)  # [[1.7], [1.0]]
```

### CODE REVIEW

File: python/paddle/__init__.py [hunk 1]
```diff
@@ -193,6 +193,7 @@
 from .tensor.math import square  # noqa: F401
 from .tensor.math import stanh  # noqa: F401
 from .tensor.math import sum  # noqa: F401
+from .tensor.math import nansum  # noqa: F401
```
Comment (jeff41404) [inline python/paddle/__init__.py:196]: public api should be add in __all__ list below

File: python/paddle/tensor/__init__.py [hunk 2]
```diff
@@ -151,6 +151,7 @@
 from .math import square  # noqa: F401
 from .math import stanh  # noqa: F401
 from .math import sum  # noqa: F401
+from .math import nansum  # noqa: F401
```
Comment (jeff41404) [inline python/paddle/tensor/__init__.py:154]: for paddle.Tensor.nansum, we should add nansum in list of tensor_method_func below

**Review by Superjomn — DISMISSED** [review]
LGTM

File: python/paddle/tensor/math.py [hunk 3]
```diff
@@ -780,6 +780,71 @@ def get_dtype(x, dtype):
     return out
 
 
+def nansum(x, axis=None, dtype=None, keepdim=False, name=None):
+    """
+    Computes the sum of tensor elements over the given dimension, treating Not a Numbers (NaNs) as zero.
```
Comment (TCChenlong) [inline python/paddle/tensor/math.py]: 需要注意一致， axis 和 dimension 下同
given dimension -> given axis

File: python/paddle/tensor/math.py [hunk 4]
```diff
@@ -780,6 +780,71 @@ def get_dtype(x, dtype):
...
+            value is False.
+        name (str, optional): The default value is None. Normally there is no need for
+            user to set this property.  For more information, please refer to :ref:`api_guide_Name`
+
+    Returns:
+        Tensor: Results of summation operation on the specified axis of input Tensor `x`,
+        if `x.dtype='int32'`, it's data type is `'int64'`, 
+        otherwise it's data type is the same as `x`.
+
+    Raises:
```
Comment (TCChenlong) [inline python/paddle/tensor/math.py]: 这个不用写

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**